app/src/main/java/app/foodfinderapp/MainActivity.kt

```diff
@@ -90,25 +91,33 @@ class MainActivity : AppCompatActivity() {
             }
         }
 
-        val restaurantRef = db.collection("restaurants")
-        restaurantRef.get()
-            .addOnSuccessListener { documents ->
-            for (document in documents) {
-                val restaurant = document.toObject(Restaurant::class.java)
-                restaurantList.add(restaurant)
+        restaurantService.getAllRestaurants(
+            onSuccess = { fetchedRestaurants ->
+                restaurantList.addAll(fetchedRestaurants)
+                adapter.notifyDataSetChanged()
+            },
+            onFailure = { exception ->
+                Log.e(TAG, "Error getting restaurants", exception)
             }
-            adapter.notifyDataSetChanged()
-        }.addOnFailureListener { exception ->
-            Log.e(TAG, "Error getting restaurants", exception)
-        }
-
+        )
     }
 
+    /**
+     * Determines if the user is logged in.
+     *
+     * @return true if the user is logged in, returns false otherwise.
+     */
```

Comment:
While JavaDoc above classes and public methods is good, this function's name is descriptive enough to indicate what is stated here, in the JavaDoc.